Realm/RLMObjectBase.mm

```diff
@@ -159,32 +156,67 @@ - (NSArray *)linkingObjectsOfClass:(NSString *)className forProperty:(NSString *
     }
     
     size_t col = prop.column;
-    NSUInteger count = _row.get_backlink_count(*table, col);
+    NSUInteger count = object->_row.get_backlink_count(*table, col);
     NSMutableArray *links = [NSMutableArray arrayWithCapacity:count];
     for (NSUInteger i = 0; i < count; i++) {
-        [links addObject:RLMCreateObjectAccessor(_realm, schema, _row.get_backlink(*table, col, i))];
+        [links addObject:RLMCreateObjectAccessor(object->_realm, schema, object->_row.get_backlink(*table, col, i))];
     }
     return [links copy];
 }
 
-- (id)objectForKeyedSubscript:(NSString *)key {
-    if (_realm) {
-        return RLMDynamicGet(self, key);
+id RLMObjectBaseObjectForKeyedSubscript(RLMObjectBase *object, NSString *key) {
+    if (!object) {
+        return nil;
+    }
+
+    if (object->_realm) {
+        return RLMDynamicGet(object, key);
```

Comment:
More importantly, you should note that `valueForKey:` does not work for dynamic objects.